keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1224 @@
+resource usage of all containers in a pod rather than predicting each container’s individual needs. This proposal aims to support pod-level resource requests 
+and limits that would allow Kubernetes to manage the total resource consumption of the pod, freeing users from the burden of meticulously allocating resources
+for each container. This simplified approach is particularly valuable for tightly coupled applications, and it will enable Kubernetes to leverage the underlying
+cgroup resource management mechanisms more effectively for better resource utilization for bursty workloads. 
+
+This new feature of resource specification at pod-level would complement, not replace, the existing container-level limits, providing users with greater flexibility.
+
+
+### Goals
+
+This proposal aims to:
+
+1. Extend Pod API to allow specifying resource limits on pod level as an opt-in feature. 
+2. Make this feature compatible with existing usage of container level resource/limits, and other features like topology manager, memory manager, etc.
+
+### Non Goals
+
+
+1. This KEP focuses on the core resource types of CPU and memory. It doesn’t intend to address other resource types (e.g. GPU, storage, network bandwidth)
+at pod level in this phase. However, it could be considered in future extensions of the KEP.
+
+
+## Proposal
+
+
+### User Stories
+
+
+#### Story 1
+
+A development environment implemented as a Kubernetes Pod allows for separation of tools and a (web-based) IDE between multiple side-cars.
+The development environment defines a container with the web-server serving the IDE itself, and constrains it to use a certain amount of memory.
+Additional tools are provided in additional side-car deployments - for example an LSP( https://langserver.org/) service, a terminal, and more. 
+Using this new feature, the containers providing the terminal, the LSP services, and the set of tools being utilized can share the resource limit
+defined for the pod. Consider the following pod definition:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    run: ide
+  name: myide
+spec:
+  resources:
+    limits:
+      memory: "1024Mi"
+      cpu: "4"
+  containers:
+  - image: shell
+    name: debian:buster
+  - image: tool1
+    name: first-tool:latest
+  - image: tool2
+    name: second-tool:latest
+  - name: ide
+    image: theia:latest
+    resources:
+      requests:
+        memory: 128M
+        cpu: "0.5"
+      limits:
+        memory: 256M
+        cpu: "1"
+```
+
+Using the pod-level resource definition enables the `tool1` and `tool2` containers to be constrained by the total limit for the pod. 
+
+Without this feature, the developer would need to decide a-priori how much resources to allocate to the tools - and this is not easy to do for this workload.
+
+
+#### Story 2
+
+Consider a scenario where a pod has some containers with occasional resource spikes while other containers often have idle resources. With current container 
```

Comment:
This story is kind of weak, and a bit misleading.  I think it needs to emphasize that it saves resources when you know that resource bursts are not coordinated.  Rather than planning for the sum of a percentile of the max of each container, you can plan for a percentile of the aggregate, which should be lower than the sum of individuals' max.

minor point

Reply:
+1 on adding this clarification. Also to make case stronger - example of service mesh sidecar can be added. If sidecar that provides the service mesh is CPU throttled, it makes the whole Pod non-functional even if it has CPU cycles. So it only logical to make service mesh share resources with the main container in many cases